daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -1941,3 +1940,58 @@ func TestExperimentalCheck(t *testing.T) {
 	require.Equal(t, 200, rec.Code)
 	require.Equal(t, "true\n", string(rec.Body.Bytes()))
 }
+
+func setupTestDevMode(devmode bool, t *testing.T) (v2.Handlers, echo.Context, *httptest.ResponseRecorder, []account.Root, []transactions.SignedTxn, func()) {
+	numAccounts := 1
+	numTransactions := 1
+	offlineAccounts := true
+	mockLedger, rootkeys, _, stxns, releasefunc := testingenv(t, numAccounts, numTransactions, offlineAccounts)
+	mockNode := makeMockNode(mockLedger, t.Name(), nil, cannedStatusReportGolden, devmode)
+	dummyShutdownChan := make(chan struct{})
+	handler := v2.Handlers{
+		Node:     mockNode,
+		Log:      logging.Base(),
+		Shutdown: dummyShutdownChan,
+	}
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	return handler, c, rec, rootkeys, stxns, releasefunc
+}
+
+func TestTimestampOffsetNotInDevMode(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	handler, c, rec, _, _, releasefunc := setupTestDevMode(false, t)
+	defer releasefunc()
+
+	// TestGetBlockTimeStampOffset 400 - offset is not set and mock node is
+	// not on dev mode
+	err := handler.GetBlockTimeStampOffset(c)
+	require.NoError(t, err)
+	require.Equal(t, 400, rec.Code)
+	// TestSetBlockTimeStampOffset 400 - cannot set timestamp offset when not
+	// in dev mode
+	err = handler.SetBlockTimeStampOffset(c, 1)
+	require.NoError(t, err)
+	require.Equal(t, 400, rec.Code)
+}
+
+func TestTimestampOffsetInDevMode(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	handler, c, rec, _, _, releasefunc := setupTestDevMode(true, t)
+	defer releasefunc()
+
+	// TestSetBlockTimeStampOffset 200
+	err := handler.SetBlockTimeStampOffset(c, 1)
```

Comment:
please also check if you get 400 if you try to get the offset before setting it.

Reply:
Based on PR adjustments, we no longer return a 400 if the node is in dev mode. Instead, the response is just a 200 with `{offset: 0}` whether it is not set or whether the user set it to 0. 

Does this sound reasonable, or should we also explicitly return a separate error if an offset was never set? I decided that they are semantically equivalent so there is less of a reason now, i.e. not setting the offset and the user setting the offset to 0 both reverts the timestamps to real time.

Reply:
The ambiguity is not great, but it's also something we can improve later without breaking the interface so I don't think this is a blocker.

Thinking out loud here... what if DevMode ALWAYS uses the offset. We can initialize it to `3` so that blocks sorta look like normal blocks. Now the getter never has to worry about the "unset" case, and if the user wants all blocks to happen at the same time they can set it to 0, which is also a valid value.

Reply:
I agree that the ambiguity is not good, so I addressed in [e7f482c](https://github.com/algorand/go-algorand/pull/5296/commits/e7f482c6d07c711389618a6330c5837fc941ef39)

If offset is nil, we default to real clock timestamps, and if it is explicitly set to 0, then the timestamp does not advance.